e2e_testing/join.py

```diff
@@ -0,0 +1,51 @@
+import pandas as pd
+import numpy as np
+import time
+import sys
+sys.path.append('../python')
+from pandas_exporter import export_pandas, annotate_pandas
+
+@export_pandas
+@annotate_pandas(
+    arg0 = {"type": "DataFrame",
+            "schema": {"a": "i32", "b": "i32"},
+            "indices": ["key", 1, "hello"],
+            "dims": [3, 2]},
+    arg1 = {"type": "DataFrame",
+            "schema": {"a": "i32", "b": "i32"},
+            "indices": ["key", 1, "hello"],
+            "dims": [3, 2]},
+)
+def compute(df0, df1):
+    """
+    This test function demonstrates how pandas-mlir can export functions
+    when provided with complete information about the input.
+
+    Args:
+        df (pd.DataFrame): DataFrame with no nullable types and
+        fixed size.
+
+    Returns:
+        A linear combination of the "a" column and "b" column.
+
+    """
+    # Perform an inner join on the 'a' column.
+    # Should return the original dataframe because we
+    return pd.merge(
```

Comment:
@harsh-nod: As a starting point (mostly just for discussion). I just copied over the original `dataframe.py` file as a starting point. Happy to refactor / migrate over to a different testing strategy if there is a preference for something else.